web/connect/result/handler.go

```diff
@@ -0,0 +1,82 @@
+package result
+
+import (
+	"html/template"
+	"net/http"
+	"strconv"
+
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type handler struct {
+	connections  sdkservices.Connections
+	integrations sdkservices.Integrations
+}
+
+func NewHandler(s sdkservices.Services) handler {
+	return handler{connections: s.Connections(), integrations: s.Integrations()}
+}
+
+// ServeHTTP is an HTTP handler that serves generic
+// connection success and error webpages for all integrations.
+func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request, t *template.Template, ok bool) {
+	// Read the connection ID from the request path.
+	cid, err := sdktypes.StrictParseConnectionID(r.PathValue("id"))
+	if err != nil {
+		http.Error(w, "Bad connection ID", http.StatusBadRequest)
+		return
+	}
+
+	// Fetch the connection details.
+	conn, err := sdkerrors.IgnoreNotFoundErr(h.connections.Get(r.Context(), cid))
+	if err != nil {
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	if !conn.IsValid() {
+		http.Error(w, "Connection not found", http.StatusNotFound)
+		return
+	}
+
+	// Fetch the connection's integration details.
+	integ, err := h.integrations.GetByID(r.Context(), conn.IntegrationID())
```

Comment:
don't we have getIntegrationByConnectionID? do we really need to fetch the connection just to get the integration?

Reply:
We don't, afaik. Also, I'm not sure it would belongs in the integration service, because it couples it to the connections service. But it's an idea to think about in a separate PR.

Reply:
opened a ticket ENG-1242
resolving this for now